### ISSUE TITLE
Why does the background of each char in the synthesis images look like inconsistent? 

### ISSUE THREAD

**Issue by songkq** [ganji15/HiGANplus#3]
Hi, @ganji15 thanks for your contribution. 
As shown in the picture, `"Sweat"` is the real style input image, and the others are synthesis images using the pretrained model.
I'm confusing about  the  inconsistent background of each char in the synthesis images. 
Also, it looks like that the style transfer failed to imitate that of the input image. The synthesis chars look like some styles in the `IAM` trainset. I'm wondering how to improve the style transfer ability for HiGAN+ in a few-shot task.
By the way, can I use the model pretrained on word-level dataset for generating images on text lines of sentences?
Could you please give some kind advide?

![image](https://user-images.githubusercontent.com/30183023/225569070-3940ff2f-cf21-4e70-8283-de785305079f.png)

**Comment by ganji15** [comment]
Hi, @songkq 

I think the background of the given picture "sweat" is slightly complex and this kind of background may rarely appear in the IAM dataset, therefore,  it is difficult to imitate those rare styles (that have been seldom seen before during training) for HiGAN+. 

It also should be noted that handwriting style transfer is challenging since humans' handwriting is very arbitrary, and the reported style transfer accuracy on the test set (i.e., unseen styles) of IAM  is only ~58% (as reported in Table 1 of our paper). Therefore, some novel techniques or strategies are necessary to improve the style transfer ability for HiGAN+.

Lastly, the model pre-trained on a word-level dataset can be directly used for generating images on text lines of sentences  (1. if the background is clean OR 2. if world-level samples contain many short phrases with spaces).

Thanks!

**Comment by songkq** [comment]
@ganji15 Thanks. I'm wondering how many training samples are needed at least for imitating a  new writer's style the during finetuning.  Have you try to train a HiGAN+ model based on the IAM line-level dataset? How about the ability of style transfer compared with that of the word-level model?
As you mentioned that "some novel techniques or strategies are necessary to improve the style transfer ability for HiGAN+.", could you please explain more details about these strategies. For example, will self-supervised learning be helpful?

**Comment by ganji15** [comment]
Hi, @songkq 

1. I haven't conducted any experiments to verify how many training samples are needed at least, but I guess that more training samples will definitely lead to better style transfer results.

2. The model pre-trained on a word-level dataset can be used for generating text lines, and it perhaps can get better results by fine-tuning the pre-trained world-level model with line-level data.  You can prepare the line-level data to conduct some quick experiments.

3.  I am sorry that I have not come up with any good ideas yet, but I think both self-supervised learning and contrastive learning are worth to be considered.

Thanks!

**Comment by hmppt** [comment]
> 你好，@ganji15感谢您的贡献。 如图所示，`"Sweat"`是真实风格的输入图像，其他是使用预训练模型的合成图像。 我对合成图像中每个字符的背景不一致感到困惑。 此外，看起来风格迁移未能模仿输入图像的风格迁移。合成字符看起来像`IAM`训练集中的某些样式。我想知道如何在少数任务中提高 HiGAN+ 的风格迁移能力。 顺便说一句，我可以使用在单词级数据集上预训练的模型来生成句子文本行上的图像吗？ 你能给点建议吗？
> 
> ![图像](https://user-images.githubusercontent.com/30183023/225569070-3940ff2f-cf21-4e70-8283-de785305079f.png)

嗨，您好！
我想请问一下你示例中的sweat经过什么样的变换吗？我用手机拍摄了几张我自己的手写字体，并resize到64*16（len（text）），然后使用gen_h5file生成测试所需要的文件。如下所示，左边是我的手写字体，右边是HiGAN+生成的字体。很显然，HiGAN+生成的字体与我的字体并不一样（包括“HaHa”的倾斜角度）。
我的疑问是：是什么导致了这样的结果，为什么我的字体背景是黑色的，我该如何做呢？
![image](https://user-images.githubusercontent.com/43752397/227761963-328a461b-6f91-49b9-aa5e-8ecea01b4031.png)

@ganji15

**Comment by ganji15** [comment]
@hmppt 
Since the writing styles of your given samples are **significantly different** from that of IAM dataset, the pre-trained HiGAN+ model on IAM dataset fails to imitate such styles due to the distribution gap. So, you must train a new HiGAN+ model using your prepared handwriting samples.

**Comment by hmppt** [comment]
> @hmppt Since the writing styles of your given samples are **significantly different** from that of IAM dataset, the pre-trained HiGAN+ model on IAM dataset fails to imitate such styles due to the distribution gap. So, you must train a new HiGAN+ model using your prepared handwriting samples.

Thanks for your answer. I have another question about IAM datasets: 
![image](https://user-images.githubusercontent.com/43752397/227868304-428dd3e9-2d16-4e92-887e-b2aaf2799144.png)
As shown in the figure above, this is part of the data in the IAM words dataset. What does the name of each picture mean? Such as the "a01", "000u", "00", which one corresponds to the "wids" in the code?

**Comment by songkq** [comment]
@ganji15 Hi, when I carry out some experiments based on text lines, HiGAN can work well on the lowercase except the character `"w"`. I'm confusing why it doesn't work on `"w"` since it also occupies a large mount of the training dataset. Could you please give some advice?

**Comment by 13209413223** [comment]
您好！能介绍一下数据中的这七个文件的构建方法吗？谢谢
